go/libraries/doltcore/doltdb/foreign_key_coll.go

```diff
@@ -203,7 +203,7 @@ func (fk ForeignKey) ValidateTableSchema(sch schema.Schema) error {
 			return fmt.Errorf("foreign key `%s` has entered an invalid state, table `%s` has unexpected schema", fk.Name, fk.TableName)
 		}
 	}
-	if !sch.Indexes().Contains(fk.TableIndex) {
+	if (fk.TableIndex != "" && !sch.Indexes().Contains(fk.TableIndex)) || (fk.TableIndex == "" && sch.GetPKCols().Size() == 0) {
```

Comment:
Maybe want to verify that the primary key length is `>=` the column count, rather than just checking that there are primary keys. Something like `sch.GetPKCols().Size() >= len(fk.TableColumns)`